pkgs/development/web/nodejs/npm-patches.nix

```
@@ -0,0 +1,23 @@
+{ fetchpatch }:
+
+[
+  # Makes `npm pack` obey `--foreground-scripts`
+  (fetchpatch {
+    name = "libnpmpack-obey-foreground-scripts.patch";
+    url = "https://github.com/npm/cli/commit/e4e8ae20aef9e27e57282e87e8757d5b364abb39.patch";
+    hash = "sha256-XjY0UCI95mpMJLvpWk2Oy39EcYDHYNSrpwfm/4vviXI=";
+    extraPrefix = "deps/npm/";
+    stripLen = 1;
+    includes = [ "deps/npm/workspaces/libnpmpack/lib/index.js" ];
+  })
+
+  # Makes `npm pack` obey `--ignore-scripts`
+  (fetchpatch {
+    name = "libnpmpack-obey-ignore-scripts.patch";
+    url = "https://github.com/npm/cli/commit/a990c3c9a0e67f0a8b6454213675e159fe49432d.patch";
+    hash = "sha256-PTp2o64CqsEJz5/1yaW6DGBy95/1QqRfyX7PHm4pzN4=";
+    extraPrefix = "deps/npm/";
+    stripLen = 1;
+    includes = [ "deps/npm/workspaces/libnpmpack/lib/index.js" ];
+  })
```

Actually this does not apply cleanly because in the distributed source, the workspace dependencies only exist built in to the `deps/npm/node_modules` directory rather than in `deps/npm/workspaces`

The below at least allows the patch to succeed for all 3 versions. I'm building nodejs now to test further

```suggestion
  # Makes `npm pack` obey `--foreground-scripts`
  (fetchpatch {
    name = "libnpmpack-obey-foreground-scripts.patch";
    url = "https://github.com/npm/cli/commit/e4e8ae20aef9e27e57282e87e8757d5b364abb39.patch";
    hash = "sha256-NQ8CZBfRqAOMe0Ysg3cq1FiferWKTzXC1QXgzX+f8OU=";
    extraPrefix = "deps/npm/node_modules/";
    stripLen = 2;
    includes = [ "deps/npm/node_modules/libnpmpack/lib/index.js" ];
  })

  # Makes `npm pack` obey `--ignore-scripts`
  (fetchpatch {
    name = "libnpmpack-obey-ignore-scripts.patch";
    url = "https://github.com/npm/cli/commit/a990c3c9a0e67f0a8b6454213675e159fe49432d.patch";
    hash = "sha256-eA5YST9RxMMjk5FCwEbl1HQUpXZuwWZkx5WC4yJium8=";
    extraPrefix = "deps/npm/node_modules/";
    stripLen = 2;
    includes = [ "deps/npm/node_modules/libnpmpack/lib/index.js" ];
  })
```

---

This is what I get for submitting without building 😖 will fix, sorry, thanks!